src/coreclr/interpreter/compiler.h

```diff
@@ -586,4 +835,41 @@ class InterpCompiler
      return compiler->AllocMemPool0(sz);
  }
 
+template<typename T>
+int32_t InterpDataItemIndexMap::GetDataItemIndexForT(const T& lookup)
+{
+    VarSizedDataWithPayload<T> key;
+    key.payload = lookup;
+
+    dn_simdhash_ght_t* hash = GetHash();
+
+    void* resultAsPtr = nullptr;
+    if (dn_simdhash_ght_try_get_value(hash, (void*)&key, &resultAsPtr))
+    {
+        return (int32_t)(size_t)resultAsPtr;
+    }
+
+    // Assert that there is no padding in the struct, so a fixed size hash unaware of padding is safe to use
+    assert(sizeof(VarSizedData) == sizeof(void*));
+    assert(sizeof(T) % sizeof(void*) == 0);
+    assert(sizeof(VarSizedDataWithPayload<T>) == sizeof(T) + sizeof(void*));
```

Comment:
These should probably also be static_assert instead of assert, so that they will be checked in release builds